app/client/components/productPage.tsx

```diff
@@ -264,16 +250,32 @@ const headerCss = css({
 
 export const Membership = (props: RouteableProps) => (
   <>
-    <PageHeaderContainer>
+    <PageHeaderContainer selectedNavItem={navLinks.membership}>
       <h1 className={headerCss}>Membership</h1>
     </PageHeaderContainer>
-    <MembershipAsyncLoader
-      fetch={loadMembershipData}
-      render={renderMembershipData}
+    <MembersDatApiAsyncLoader
+      fetch={ProductTypes.membership.fetchProductDetail}
+      render={getProductRenderer(ProductTypes.membership)}
       loadingMessage="Loading your membership details..."
     />
     <PageContainer>
       <MembershipLinks />
     </PageContainer>
   </>
 );
+
+export const Contributions = (props: RouteableProps) => (
+  <>
+    <PageHeaderContainer selectedNavItem={navLinks.contributions}>
+      <h1 className={headerCss}>Contributions</h1>
+    </PageHeaderContainer>
+    <MembersDatApiAsyncLoader
+      fetch={ProductTypes.contributions.fetchProductDetail}
+      render={getProductRenderer(ProductTypes.contributions)}
+      loadingMessage="Loading details of your contributions..."
```

Comment:
I'm not sure about this copy... what about: 'Loading your recurring contribution details...'?

(Feel free to ignore if you've already been through this with UX!)

Reply:
i've actually reworked this more generally in subsequent PR (#129)